src/pydata_sphinx_theme/assets/styles/components/_switcher-theme.scss

```diff
@@ -32,13 +18,13 @@
 }
 
 html[data-mode="auto"] .theme-switch-button span[data-mode="auto"] {
-  display: flex;
+  display: inline;
```

Comment:
This needs to be inline in order for the line-height property to work. Will add comment.